### DISCUSSION TITLE
Keycloak In Docker: ERROR: Strict hostname resolution configured but no hostname setting provided

### DISCUSSION THREAD

**Discussion by archmangler** [keycloak/keycloak#20437] [Archive]
I'm following this keycloak documentation on how to get the KC optimised image running on docker:

https://www.keycloak.org/server/containers

I get the following cryptic error when trying to run the "optimized" keycloak docker image:

```
(base) welcome@Traianos-MacBook-Pro keycloak-docker % docker run --name mykeycloak -p 8443:8443 \
        -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=P@ssw0rd \
        quay.io/keycloak/keycloak:latest \
        start --optimized
ERROR: Unexpected error when starting the server in (production) mode
ERROR: Failed to start quarkus
ERROR: Strict hostname resolution configured but no hostname setting provided
For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
```

It's not clear from the documentation how to resolve this error:

```
ERROR: Strict hostname resolution configured but no hostname setting provided
```

Any idea how to resolve this?

**Comment by vmuzikar** [comment]
@archmangler Thanks for the report. Since you are starting the server in production mode (you are using `start` and not `start-dev`), you also need to specify the `--hostname` option. Please see the relevant [guide](https://www.keycloak.org/server/configuration#_starting_keycloak_in_production_mode).

**Reply by joeha11** [reply]
Any chance that the guide could get updated to include that flag in the commands listed?
I ran into the same issue today when following the same guide.

**Reply by shawkins** [reply]
@joeha11 #33539 will address that
